common/src/main/java/com/microsoft/identity/common/adal/error/AuthenticationException.java

```diff
@@ -65,6 +122,79 @@ public String getMessage() {
         return getLocalizedMessage(null);
     }
 
+    /**
+     * Gets the status code returned from http layer.
+     *
+     * @return status code from http layer. Return -1 if status code is not initialized.
+     */
+    public int getServiceStatusCode() {
+        return mServiceStatusCode;
+    }
+
+    /**
+     * Gets the response body that may be returned by the service.
+     *
+     * @return response body map, null if not initialized.
+     */
+    public HashMap<String, String> getHttpResponseBody() {
+        return mHttpResponseBody;
+    }
+
+    /**
+     * Get the response headers that indicated an error.
+     *
+     * @return The response headers for the network call, null if not initialized.
+     */
+    public HashMap<String, List<String>> getHttpResponseHeaders() {
+        return mHttpResponseHeaders;
+    }
+
+    public void setHttpResponseBody(HashMap<String, String> body) {
+        mHttpResponseBody = body;
+    }
+
+    public void setHttpResponseHeaders(HashMap<String, List<String>> headers) {
+        mHttpResponseHeaders = headers;
+    }
+
+    public void setServiceStatusCode(int statusCode) {
+        mServiceStatusCode = statusCode;
+    }
+
+    public void setHttpResponse(HttpWebResponse response) {
+        if (null != response) {
+            mServiceStatusCode = response.getStatusCode();
+
+            if (null != response.getResponseHeaders()) {
+                mHttpResponseHeaders = new HashMap<>(response.getResponseHeaders());
+            }
+
+            if (null != response.getBody()) {
+                try {
+                    mHttpResponseBody = new HashMap<>(HashMapExtensions.getJsonResponse(response));
+                } catch (final JSONException exception) {
+                    //Log.e(AuthenticationException.class.getSimpleName(), "Json exception",
+                    //        ExceptionExtensions.getExceptionMessage(exception),
+                    //        ADALError.SERVER_INVALID_JSON_RESPONSE);
+
+                    Log.e(AuthenticationException.class.getSimpleName(), ADALError.SERVER_INVALID_JSON_RESPONSE.toString(), exception);
+                }
+            }
+        }
+    }
+
+    public void setHttpResponse(int statusCode, HashMap<String, List<String>> headers, HashMap<String, String> body) {
+
+        if(body != null)
```

Comment:
As a matter of style/preference I generally avoid using brace-less `if` clauses. 

While a nitpick, it'd be my preference to have these braced; the [Android contributor guide](https://source.android.com/setup/code-style#use-standard-brace-style) advocates similarly.